### ISSUE TITLE
[AllyAudit] Dropdown will be accessible 

### ISSUE THREAD

**Issue by ssabo1** [sparkdesignsystem/Backlog#1325]
**User Story**
As Spark we want to Correct focus order for Masthead and Masthead Extended so that they are accessible.

**Notes:**
Paciello Audit on Spark Design System
QLC-15 Dropdown & Masthead 
Inaccessible custom dropdown

ISSUE DESCRIPTION
Dropdown (icon) and 17 - Dropdown there controls that when triggered reveal a list of links. Some effort has been put into marking these up as collapsible dropdown listboxes in line with the listbox pattern in the ARIA authoring practices, though the pattern is not fully implemented. The controls are incorrectly marked up as <a> elements rather than <button> elements, and there is no information conveyed about the expanded/collapsed state of the associated lists.

Masthead (extended) has 3 variants of custom dropdowns. The "Select One" control, two "Navigation Item" controls that are very similar to the 17 - Dropdown component, and one icon-based control that is very similar to 16 - Dropdown (icon). All of these controls have an incorrect role="combobox" as they do not have single-line text fields, a requirement for the combobox pattern. For all four controls, a screen reader incorrectly announces "to change the selection use the arrow keys". The only way to navigate the items in the list is by pressing the tab key. They are all lacking announced information about their expanded/collapsed state. The "Select One" control is also lacking an accessible name.

RECOMMENDATIONS
The correct ARIA pattern to be used will depend on the specific purpose of the widgets/how they are being used in practice.
For 16 - Dropdown (icon) and 17 - Dropdown, assuming that the "choices" are in fact links to different pages/locations, rather than triggers for in-page functionality, we recommend implementing a basic disclosure widget pattern (a button that opens a container with links) and removing the partial implementation of the listbox patten (i.e. the role="listbox" and role="option" attributes). The control to toggle the list of links needs to be a `<button>` element. The list of links can be hidden/displayed by toggling the CSS properties display:none and display:block on the parent `<div>` element.

For the icon control, ensure that there is an accessible name for the control either coming from the SVG or from an aria-label or aria=labelledby attribute. For demonstration purposes we've shown an aria-label. The role="option" on the list items is not necessary, nor is the aria-label on the ul.

https://www.w3.org/TR/wai-aria-practices-1.2/#keyboard-interaction-13

**AC**
- all keyboard interactions will work https://www.w3.org/TR/wai-aria-practices-1.1/#keyboard-interaction-13
- needs to use HTML structure and ARIA roles/attributes described in Paciello Audit

**Comment by ssabo1** [comment]
This is blocked by Research Issue: https://github.com/sparkdesignsystem/spark-design-system/issues/3584

**Comment by ssabo1** [comment]
**Issue:**
"My Choices" color accessibility
User Experience: Difficult to see

**Current State:**
https://angular.sparkdesignsystem.com/?path=/story/components-dropdown--informational
Currently using the black-tint-30 
same for the default
same for inside the extended masthead 
![image.png](https://images.zenhubusercontent.com/5c98da1df75c1c000158e457/210bd881-9259-4912-9554-3077574fb417)

UX to Define updated color to go in place of black-tint-30

**Comment by ssabo1** [comment]
Confirmed: This has been added to Spark Design System Team Backlog.
